api/src/main/java/org/pmiops/workbench/cdr/CdrVersionContext.java

```diff
@@ -12,7 +17,35 @@
 
   private static ThreadLocal<CdrVersion> cdrVersion = new ThreadLocal<>();
 
-  public static void setCdrVersion(CdrVersion version) {
+  private Provider<WorkbenchConfig> configProvider;
+  private FireCloudService fireCloudService;
+
+  @Autowired
+  public CdrVersionContext(Provider<WorkbenchConfig> configProvider,
+      FireCloudService fireCloudService) {
+    this.configProvider = configProvider;
+    this.fireCloudService = fireCloudService;
+  }
+
+  public void setCdrVersion(CdrVersion version) {
+    if (configProvider.get().firecloud.enforceRegistered) {
+      // TODO: map data access level to authorization domain here (RW-943)
+      String authorizationDomain = configProvider.get().firecloud.registeredDomainName;
+      if (!fireCloudService.isUserMemberOfGroup(authorizationDomain)) {
```

Comment:
I'm not sure I like encoding this business logic at this level. Up until this point, this class was pretty agnostic to where it was running and who was calling it, this brings in a bunch of deps as well.

A few other questions about this:
- How will this interact with the default CDR version? Is it possible some of those usages would sidestep auth domain enforcement?
- How will this work with DataBrowser, would they  just call the noAuthDomain variant?

Reply:
As discussed over Slack: I created a separate CdrVersionService which does this logic.

The default CDR version only indicates what CDR version to use when e.g. creating new workspaces; you still have to set the CDR version in CdrVersionContext before you can read from it.

Data browser doesn't presently use CdrVersionContext, but in future it can (without using CdrVersionService or FireCloud.)